domains/authorization/Proxy.IntegrationTests/Setup/ProxyTestAuthHandler.cs

```diff
@@ -0,0 +1,37 @@
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using EnergyOrigin.TokenValidation.b2c;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Net.Http.Headers;
+
+namespace Proxy.IntegrationTests.Setup;
+
+public class ProxyTestAuthHandler(
+    IOptionsMonitor<AuthenticationSchemeOptions> options,
+    ILoggerFactory logger,
+    UrlEncoder encoder)
+    : AuthenticationHandler<AuthenticationSchemeOptions>(options, logger, encoder)
+{
+    protected override Task<AuthenticateResult> HandleAuthenticateAsync()
+    {
+        var jwtSecurityTokenHandler = new JwtSecurityTokenHandler();
+        var authorizationHeader = Context.Request.Headers[HeaderNames.Authorization].ToString().Substring(7);
+        var securityToken = jwtSecurityTokenHandler.ReadToken(authorizationHeader) as JwtSecurityToken;
+
+        // Check for the existence of a claim that only B2C should provide
+        if (!securityToken!.Claims.Any(c => c.Type == ClaimType.OrgIds))
```

Comment:
This should be removed. In the case of an client that needs to invoke the receive-slice endpoint, which should be an open endpoint, this check would fail since it expects a client using B2C as its identity provider.

Reply:
You made this code :D

Reply:
I have fixed this. By making check if it requires auth or not.